core/packetgraph/utils/mac-table.h

```diff
@@ -193,4 +199,86 @@ static inline void *pg_mac_table_ptr_get(struct pg_mac_table *ma,
 	return ma->ptrs[part1]->entries[part2];
 }
 
+static inline void pg_mac_table_iterator_next(struct pg_mac_table_iterator *it)
+{
+	struct pg_mac_table *ma = it->table;
+	uint32_t part1 = pg_mac_table_part1(it->key);
+	uint32_t part2 = pg_mac_table_part2(it->key);
+	int first_mask = pg_mac_table_mask_idx(part1);
+	int first_sub_mask = pg_mac_table_mask_idx(part2);
+	uint64_t unset;
+	uint64_t sub_unset;
+
+	unset = ~((ONE64 << (pg_mac_table_mask_pos(part1) + 1)) - ONE64);
+	sub_unset = ~((ONE64 << (pg_mac_table_mask_pos(part2) + 1)) - ONE64);
+
+	for (int i = first_mask; i < (PG_MAC_TABLE_MASK_SIZE); ++i) {
+		if (!ma->mask[i])
+			continue;
+
+		PG_FOREACH_BIT(ma->mask[i] & unset, i2) {
+			struct pg_mac_table_ptr *sub_ma;
+
+			sub_ma = ma->ptrs[(i * 64) + i2];
```

Comment:
`sub_ma` is always used with `sub_ma->mask[j]`
You can set sub_ma like: `sub_ma = ma->ptrs[(i * 64) + i2]->mask;` and use directly sub_ma[j]